File: frontend/ContentView.js
import React, { useState } from 'react';
import { View, StyleSheet } from 'react-native';
import CustomTabBar from './CustomTabBar'; // Import the CustomTabBar
import HomeView from './HomeView';
import SettingsView from './SettingsView';
import AnalyticsView from './AnalyticsView';
import LearnView from './LearnView';
import MoreView from './MoreView';

const ContentView = () => {
    const [selectedTab, setSelectedTab] = useState(2); // Default to Home

    return (
        <View style={styles.container}>
            {/* Switch between views based on the selected tab */}
            {selectedTab === 0 && <SettingsView />}
            {selectedTab === 1 && <AnalyticsView />}
            {selectedTab === 2 && <HomeView />}
            {selectedTab === 3 && <LearnView />}
            {selectedTab === 4 && <MoreView />}

            <CustomTabBar selectedTab={selectedTab} setSelectedTab={setSelectedTab} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        paddingBottom: 20,
    },
});

export default ContentView;

File: frontend/wealthwise_ai/ContentView.swift
//
//  ContentView.swift
//  wealthwise_ai
//
//  Created by Allison Brown on 9/28/24.
//

import SwiftUI

struct ContentView: View {
    @State private var selectedTab = 2 // Default to Home

    var body: some View {
        VStack {
            Spacer()
            // Switch between views based on the selected tab
            switch selectedTab {
            case 0:
                SettingsView()
            case 1:
                AnalyticsView()
            case 2:
                HomeView()
            case 3:
                LearnView()
            case 4:
                MoreView()
            default:
                HomeView() // Fallback to Home
            }
            CustomTabBar(selectedTab: $selectedTab) // Bind the selected tab
        }
        .padding(.bottom, 20) // Add some padding to the bottom
    }
}

struct ContentView_Previews: PreviewProvider {
    static var previews: some View {
        ContentView()
    }
}
